lib/trento/application/usecases/sap_systems/sap_systems.ex

```diff
@@ -52,4 +59,63 @@ defmodule Trento.SapSystems do
     |> where([d], d.host_id == ^host_id)
     |> Repo.all()
   end
+
+  @spec deregister_application_instance(Ecto.UUID.t(), Ecto.UUID.t(), String.t(), DateService) ::
+          :ok | {:error, :instance_present} | {:error, :application_instance_not_registered}
+  def deregister_application_instance(
+        sap_system_id,
+        host_id,
+        instance_number,
+        date_service \\ DateService
+      ) do
+    case Repo.get_by(ApplicationInstanceReadModel,
+           sap_system_id: sap_system_id,
+           host_id: host_id,
+           instance_number: instance_number
+         ) do
+      %ApplicationInstanceReadModel{absent_at: nil} ->
+        {:error, :instance_present}
+
+      _ ->
```

Comment:
Yes, as you say. The not existing scenario is handled by the domain.
If `absent_at` is nil we cannot deregister. Otherwise we deregister (if the instance is there)